subsys/bootloader/CMakeLists.txt

```diff
@@ -171,19 +171,92 @@ include (${CMAKE_CURRENT_LIST_DIR}/cmake/debug_keys.cmake)
 
 set(sign_depends kernel_elf ${key_file_depends})
 
+set(hash_file firmware.sha256)
+set(signature_file firmware.signature)
+set(public_key_file public.pem)
+
+set(hashcmd
```

Comment:
Is there any reason we might want to use OpenSSL for this as well?

Edit: I guess this is not needed when openssl is used, and its here only because the custom command might need it as well?

In that case, would it still make sense to move this to under the `if PYTHON ECDSA` as someone wants to use their own signing probably also wants to use their own hashing?

Reply:
> Is there any reason we might want to use OpenSSL for this as well?

Not really. There's nothing particularly sensitive about this step compared to other building steps. hashlib is a built-in Python library, so I think it should be safe. My thinking was that I'd provide options just for the part that handles private keys.

> Edit: I guess this is not needed when openssl is used, and its here only because the custom command might need it as well?

It's used by all three, `openssl` as well.